src/arrayMethodReduce.js

```diff
@@ -6,6 +6,19 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    let accumulator = initialValue;
+    let strartIndex = 0;
+
+    if (arguments.length === 1) {
+      accumulator = this[0];
+      strartIndex = 1;
+    }
+
+    for (let i = strartIndex; i < this.length; i++) {
```

Comment:
U have created the variable (`strartIndex`) for index. So u can rewrite your `for` like `for (; startIndex <  this.length; startIndex++)`
but it's up to u.